zoning-toolkit-ui/src/mods/zoning-toolkit-button.tsx

```diff
@@ -1,78 +1,53 @@
-﻿import { ModuleRegistryExtend } from "cs2/modding";
-import { Button, FloatingButton, Tooltip } from "cs2/ui";
-import React from 'react';
-import iconStyles from "./icon.module.scss";
-import icon from "../../assets/Zoning Toolkit Icon.png";
-import { useModUIStore, withStore } from "./state";
-import Draggable from "react-draggable";
-
-interface ButtonState {
-	isHovered: boolean
-}
-
-class ButtonComponentInternal extends React.Component<{}, ButtonState> {
-	constructor(props: {}) {
-		super(props);
-		// Initialize state
-		this.state = { isHovered: false };
-	}
+﻿import { Button } from "cs2/ui";
+import React, { CSSProperties } from 'react';
 
-	// Method to handle mouse enter
-	handleMouseEnter = () => {
-		console.log("Mouse on button.")
-		this.setState({ isHovered: true });
-	};
+import menuIcon from "../../assets/icons/menu_icon.svg";
+import menuButtonStyles from "./zoning-toolkit-button.module.scss";
+import { useModUIStore, withStore } from "./state";
+import VanillaBindings from "./vanilla-bindings";
+import { getModeFromString, zoneModeIconMap } from "./utils";
 
-	// Method to handle mouse leave
-	handleMouseLeave = () => {
-		console.log("Mouse off button.")
-		this.setState({ isHovered: false });
-	};
+const { DescriptionTooltip } = VanillaBindings.components;
 
-	// Method to handle click
-	handleMouseClick = () => {
-		console.log("Mouse clicked button.");
+class ZoningToolkitMenuButtonInternal extends React.Component {
+	handleMenuButtonClick = () => {
+		console.log("ZT Mouse clicked toolkit menu button");
 		useModUIStore.getState().updateUiVisible(!useModUIStore.getState().uiVisible)
 	}
 
 	render() {
 		const photomodeActive = useModUIStore.getState().photomodeActive
 
-		let style: React.CSSProperties = {
-			position: "relative",
-			zIndex: 100,
-			pointerEvents: "auto",
-			borderRadius: "50%",
-			border: "2px solid black"
-		};
-
-		const hoverStyle: React.CSSProperties = {
-			...style,
-			border: '2px solid white',
-		}
+		// DEBUG: Can change to show zone indicator in tool menu icon (a little cramped though...)
+		const showZoneIndicatorInIcon = false;
+		const currentZoningMode = getModeFromString(useModUIStore.getState().zoningMode);
+		const currentZoningModeIcon = currentZoningMode ? zoneModeIconMap[currentZoningMode] : undefined;
```

Comment:
See PR description reference about this being a little cramped... This can be removed (along with related styles) if it isn't helpful/kept.